smack-core/src/main/java/org/jivesoftware/smack/packet/IQ.java

```diff
@@ -194,7 +194,6 @@ private void appendInnerXml(XmlStringBuilder xml) {
         if (type == Type.error) {
             // Add the error sub-packet, if there is one.
             appendErrorIfExists(xml);
-            return;
```

Comment:
Please see https://discourse.igniterealtime.org/t/smack-4-4-6-iq-error-stanza-generation-does-not-allow-adding-of-an-extension-element/92622/3?u=flow and adjust the code accordingly.

Reply:
Strike that, I was being stupid.